annotate.el

```diff
@@ -782,30 +802,54 @@ to 'maximum-width'."
         ;; put each annotation on its own line
         (dolist (ov overlays)
-          (cl-incf annotation-counter)
-          (let ((face           (if (= (cl-rem annotation-counter 2) 0)
-                                    'annotate-annotation
-                                  'annotate-annotation-secondary))
-                (face-highlight (if (= (cl-rem annotation-counter 2) 0)
-                                    'annotate-highlight
-                                  'annotate-highlight-secondary)))
+          (let* ((face                (if (= (cl-rem annotation-counter 2) 0)
+                                          'annotate-annotation
+                                        'annotate-annotation-secondary))
+                 (face-highlight      (if (= (cl-rem annotation-counter 2) 0)
+                                          'annotate-highlight
+                                        'annotate-highlight-secondary))
+                 (annotation-long-p   (> (string-width (overlay-get ov 'annotation))
+                                         annotate-annotation-max-size-not-place-new-line))
+                 (position            annotate-annotation-position-policy)
+                 (position-new-line-p (cl-case position
+                                        (:new-line
+                                         t)
+                                        (:by-length
+                                         annotation-long-p)
+                                        (otherwise
+                                         nil)))
+                 (splitted-annotation (if position-new-line-p
+                                          (list (overlay-get ov 'annotation))
+                                        (save-match-data
+                                          (split-string (annotate-lineate (overlay-get ov
+                                                                                       'annotation)
+                                                                          (- eol bol))
+                                                        "\n"))))
+                 (annotation-stopper  (if position-new-line-p
+                                          (if (= annotation-counter
+                                                 (length overlays))
+                                              "\n"
+                                            "")
+                                        "\n")))
+            (cl-incf annotation-counter)
             (overlay-put ov 'face face-highlight)
-            (dolist (l (save-match-data
-                         (split-string (annotate-lineate (overlay-get ov 'annotation)
-                                                         (- eol bol))
-                                       "\n")))
+            (when position-new-line-p
+              (setf prefix-first " \n"))
+            (dolist (l splitted-annotation)
               (setq text
```

Comment:
It feels like `text` is to generic a variable name for its scope. I think I'd either make `text` much more local, or give it a more descriptive name. (In general I try to use short names only very locally, where their meaning is clear from context).

Reply:
Hi!

I agree text is a bit generic, so I changed it with `annotated-text` t, is it OK for you?

Bye!!
C.